src/features/user/infrastructure.rs

```diff
@@ -42,35 +61,60 @@ pub(crate) async fn user_seen(
     )
     .bind(client_hash)
     .execute(&mut *pool)
-    .await?;
+    .await
+    {
+        Ok(r) => r,
+        Err(error) => {
+            tracing::error!("{error:?}");
+            return Err(UserError::InvalidUserId);
+        }
+    };
 
     Ok(result.rows_affected() == 1)
 }
 
 pub(crate) async fn delete_user_by_client_hash(
     app_ctx: &AppContext,
     client_hash: &str,
-) -> Result<u64, sqlx::Error> {
-    let mut pool = app_ctx.infrastructure().get_repository().await?;
-
-    let rows_deleted = sqlx::query(
+) -> Result<u64, UserError> {
+    let mut pool = match app_ctx.infrastructure().get_repository().await {
+        Ok(p) => p,
+        Err(error) => {
+            tracing::error!("{error:?}");
+            return Err(UserError::FailedToDeleteUserRecord);
+        }
+    };
```

Comment:
This pattern seems common, could we provide a HOF or something to make this less verbose?

Reply:
Go ahead and merge when you've made these changes 🙏